### ISSUE TITLE
ZC1085 false positive + harmful autofix: unquoted for-loop expansions are correct Zsh

### ISSUE THREAD

**Issue by afadesigns** [afadesigns/zshellcheck#1365]
### Summary
ZC1085 ("Quote variable expansions in `for` loops") flags `for x in $var`, and its autofix wraps the expansion in double quotes. In native Zsh this is harmful: iterating an unquoted array is the correct idiom, and quoting collapses the array into a single word — the autofix actively breaks correct code.

### Reproduction
```zsh
items=(a b c)
for x in $items; do print -r -- "$x"; done   # iterates a, b, c
```
ZC1085 warns "Unquoted variable expansion in for loop. This will split on IFS." Its autofix produces `for x in "$items"`, which iterates once over `a b c`.

### Why it is wrong
- Zsh does not word-split unquoted parameter expansions by default (`SH_WORD_SPLIT` is off; `man zshexpn` splitting rule requires it). The "splits on IFS" premise is false in native Zsh.
- Quoting an array expansion joins its elements with `IFS[1]` (`man zshparam`), so `"$items"` is one word — the autofix breaks element iteration.
- `for x in $array` (unquoted) is the standard, recommended Zsh pattern; this is the headline Zsh-vs-Bash difference. The kata imports the Bash rule into Zsh.

### Expected
Do not flag unquoted variable expansions in a `for` word list, and remove the autofix. (A possible repurpose: flag the genuinely harmful `for x in "${arr[*]}"`, which always joins.)

### Environment
- zshellcheck v1.1.0
